charts/argo-workflows/templates/_helpers.tpl

```diff
@@ -120,3 +120,19 @@ Return the target Kubernetes version
 {{- define "argo-workflows.kubeVersion" -}}
   {{- default .Capabilities.KubeVersion.Version .Values.kubeVersionOverride }}
 {{- end -}}
+
+{{/*
+Return the appropriate resource name for workflow role to patch HTTP/plugin templates
+*/}}
+{{- define "argo-workflows.workflowRole.httpTemplatePatchResourceName" -}}
+{{- $tagVersion := (trimPrefix "v" .Chart.AppVersion) -}}
+{{- if (regexMatch "^v([0-9]\\.){2}([0-9]){1}.*" .Values.controller.image.tag) -}}
+    {{- $tagVersion = (regexFind "([0-9]\\.){2}([0-9]){1}" .Values.controller.image.tag) -}}
+{{- end -}}
+
+{{- if or (semverCompare ">=3.3.0" $tagVersion) (regexMatch "^latest.*" .Values.controller.image.tag) -}}
+    {{- print "workflowtasksets/status" }}
+{{- else -}}
+    {{- print "workflowtasksets" }}
+{{- end -}}
```

Comment:
It was difficult to cover all of cases, but I think below is enough . :innocent:
- when
  - `.Values.controller.image.tag` is default (`""`)
  - `.Values.controller.image.tag` >= v3.3.0, v3.3.0-xxxxx
  - `.Values.controller.image.tag` == latest

```yaml
- apiGroups:
    - argoproj.io
  resources:
    - workflowtasksets/status
  verbs:
    - patch
```

- when
  - image < v3.3.0

```yaml
- apiGroups:
    - argoproj.io
  resources:
    - workflowtasksets
  verbs:
    - patch
```